src/global.ts

```diff
@@ -6,45 +6,127 @@ function computedState(store: any, prop: string) {
 	return computed(() => store.state[prop]);
 }
 
+/**
+ * Get the vuex store from the root vue instance
+ *
+ * @example
+ * const store = useStore();
+ */
 export function useStore<TState = any>() {
 	return getStoreFromInstance() as Store<TState>
 }
 
+/**
+ * Gets the root state of the vuex store.
+ *
+ * Injects the vuex store from the root vue instance.
+ * @param map An array of known Root State keys
+ *
+ * @example
+ * const { foo, bar } = useState(['foo', 'bar']);
+ */
+export function useState<TState>(map: KnownKeys<TState>[]): RefTypes<TState>
+/**
+ * Gets the root State of the provided vuex store.
+ * @param store An instance of the vuex store to get the state from
+ * @param map An array of known Root State keys
+ *
+ * @example
+ * const { foo, bar } = useState(store, ['foo', 'bar']);
+ */
+export function useState<TState>(store: Store<TState>, map: KnownKeys<TState>[]): RefTypes<TState>
 export function useState<TState = any>(storeOrMap: Store<TState> | KnownKeys<TState>[], map?: KnownKeys<TState>[]): RefTypes<TState> {
 	let store = storeOrMap;
 
 	if (arguments.length === 1) {
 		map = store as KnownKeys<TState>[];
 		store = getStoreFromInstance();
 	}
-	return useMapping(store, null, map, computedState);
+	return useMapping(store as Store<TState>, null, map, computedState);
```

Comment:
Using "as" in our code isn't safe enough.
It would be a problem to deliver it to other users.

Reply:
same for the next appearances of it.

Reply:
can you explain what the issue with `as` is for your codebase?
because the field store is a union of two known types this is the only way to narrow the type down without adding significant LOC. It is less _type_ safe to not narrow the type.

Reply:
the "getStoreFromInstance" is returning this type. if this function will somehow return a different type, this interface should fail.